docs/sources/fundamentals/architecture/rings.md

```diff
@@ -4,4 +4,87 @@ weight: 40
 ---
 # Consistent Hash Rings
 
-TODO: describe why and how consistent hash rings are used.
+[Consistent hash rings](https://en.wikipedia.org/wiki/Consistent_hashing)
+are incorporated into Loki cluster architectures to
+
+- aid in the sharding of log lines
+- implement high availability 
+- ease the horizontal scale up and scale down of clusters.
+There is less of a performance hit for operations that must rebalance data.
+
+Hash rings connect instances of a single type of component when 
+
+- there are a set of Loki clusters in monolithic deployment mode
+- there are multiple read components or multiple write components in
+simple scalable deployment mode
+- there are multiple instances of one type of component in microservices mode
+
+Not all Loki components are connected by hash rings.
+These components need to be connected into a hash ring:
+
+- distributors
+- ingesters
+- query schedulers
+- compactors
+- rulers
+
+In an architecture that has three distributors and three ingestors defined,
+the hash rings for these components connect the instances of same-type components.
+
+![distributor and ingester rings](../ring-overview.png)
+
+Each node in the ring represents an instance of a component.
+Each node has a key-value store that holds communication information
```

Comment:
```suggestion
Each node has a key-value pair in the store that holds communication information
```

Reply:
I'm leaving this as is.  My prose works top down, and it states that each node in the ring has a store.  It is also true that each node is represented as a key-value pair within the store, but that isn't what I wished to convey.